src/journal-remote/journal-header-util.c

```diff
@@ -0,0 +1,98 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "escape.h"
+#include "journal-header-util.h"
+#include "string-util.h"
+#include "strv.h"
+
+/* HTTP header name can contains:
+- Alphanumeric characters: a-z, A-Z, and 0-9
+- The following special characters: - and _ */
+#define VALID_HEADER_NAME_CHARS \
+        ALPHANUMERICAL "_-"
+
+#define VALID_HEADER_NAME_LENGTH 40
+
+#define VALID_HEADER_VALUE_CHARS \
+        ALPHANUMERICAL "_ :;.,\\/'\"?!(){}[]@<>=-+*#$&`|~^%"
+
+static bool header_name_is_valid(const char *e) {
+        if (!e)
+                return false;
+
+        if (strlen(e) == 0 || strlen(e) > VALID_HEADER_NAME_LENGTH)
+                return false;
+
+        return in_charset(e, VALID_HEADER_NAME_CHARS);
+}
+
+static bool header_value_is_valid(const char *e) {
+        if (!e)
+                return false;
+
+        return in_charset(e, VALID_HEADER_VALUE_CHARS);
+}
+
+int header_put(OrderedHashmap **headers, const char *name, const char *value) {
+        assert(headers);
+
+        if (!header_value_is_valid(value))
+                return -EINVAL;
+
+        if (!header_name_is_valid(name))
+                return -EINVAL;
+
+        return string_strv_ordered_hashmap_put(headers, name, value);
+}
+
+int config_parse_header(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        OrderedHashmap **headers = ASSERT_PTR(data);
+        _cleanup_free_ char *unescaped, *t;
```

Comment:
unescaped needs to be initialized to NULL, and t cannot have _cleanup_ so it needs to be on a different declaration line